cctp_primitives/src/proving_system/verifier/certificate.rs

```diff
@@ -0,0 +1,52 @@
+use crate::{
+    type_mapping::{FIELD_SIZE, FieldElement, FieldHash},
+    proving_system::{
+        verifier::{UserInputs, ZendooVerifier},
+        error::ProvingSystemError,
+    },
+    utils::serialization::SerializationUtils
+};
+use primitives::FieldBasedHash;
+use crate::utils::get_wcert_sysdata_hash;
+
+/// All the data needed to reconstruct the aggregated input for the NaiveThresholdSignatureCircuit
+/// included in the Certificate.
+pub struct CertificateProofUserInputs<'a> {
+    curr_cumulative_sc_tx_comm_tree_root: &'a [u8; FIELD_SIZE],
+    custom_fields:                        &'a [[u8; FIELD_SIZE]],
```

Comment:
We need to think how to map customFieldElements and CompresedBitVectors to the `custom_fields`. Seems better to keep them separate on the interface level, and do mapping logic inside.

Reply:
We still need to finalize the interfaces. Please take a look at the doc I shared with you too